src/connectors/gcp/auth.rs

```diff
@@ -39,8 +40,21 @@ pub(crate) async fn json_api_client(extra_headers: &HeaderMap) -> Result<Client>
     for header in extra_headers {
         headers.append(header.0, header.1.clone());
     }
-
     Ok(reqwest::Client::builder()
         .default_headers(headers)
         .build()?)
 }
+
+pub(crate) async fn merge_gcp_headers(
+    extra_headers: &HashMap<String, String>,
+) -> Result<HashMap<String, String>> {
+    let bearer = authenticate_bearer().await?;
+    let mut headers: HashMap<String, String> = halfbrown::HashMap::new();
+    headers.insert("authorization".to_string(), bearer);
+    headers.insert("content-type".to_string(), "application/json".to_string());
```

Comment:
We should probably not set the content type here and leave it up for the user/service to decide what content type they need or is on a "must-have" for all GCP services?

Reply:
this is what existed in the other implementation for headers, let me see if I can find a doc.

Reply:
👍 I'm not sure to be honest, but setting headers can cause trouble for people if they don't know :D so better be safe then sorry I think

Reply:
yeah I can take em out!